src/main/java/org/folio/service/invoice/POLInvoiceLineRelationService.java

```diff
@@ -106,6 +104,22 @@ private static org.folio.rest.acq.model.invoice.FundDistribution convertToInvoic
       .withDistributionType(org.folio.rest.acq.model.invoice.FundDistribution.DistributionType.fromValue(fundDistribution.getDistributionType().value()));
   }
 
+  private void copyAmountsAndRecalculateNewEncumbrance(List<Transaction> forCreate, List<Transaction> forDelete) {
+    BigDecimal amountExpended = forDelete.stream().map(encumbrance -> encumbrance.getEncumbrance().getAmountExpended())
+      .map(BigDecimal::valueOf).reduce(BigDecimal.ZERO, BigDecimal::add);
+    BigDecimal amountAwaitingPayment = forDelete.stream().map(encumbrance -> encumbrance.getEncumbrance().getAmountAwaitingPayment())
+      .map(BigDecimal::valueOf).reduce(BigDecimal.ZERO, BigDecimal::add);
+
+    forCreate.stream().findFirst().ifPresent(transaction -> {
+      Encumbrance encumbrance = transaction.getEncumbrance();
+      BigDecimal encumbranceAmount = BigDecimal.valueOf(encumbrance.getInitialAmountEncumbered())
+        .subtract(amountExpended).subtract(amountAwaitingPayment).max(BigDecimal.ZERO);
+      transaction.withAmount(encumbranceAmount.doubleValue())
+        .withEncumbrance(encumbrance.withAmountExpended(amountExpended.doubleValue()).withAmountAwaitingPayment(amountAwaitingPayment.doubleValue())
+          .withStatus(encumbranceAmount.compareTo(BigDecimal.ZERO) == 0 ? Encumbrance.Status.RELEASED : Encumbrance.Status.UNRELEASED));
```

Comment:
Do we have these formulas in some common place to reuse it?

Reply:
Good point. Executed to separate method. Thank you!